### ISSUE TITLE
[Suggestion] Speed Control

### ISSUE THREAD

**Issue by ZachZent** [tsunamayo/Skywanderers#1556]
I've seen this requested before, but with the speed buff for most ships it is needed more then ever. We used to have the ability to slow down and speed up when flying rather then just cruising at 100% or 0% thrust. Even with 6dof you accelerate instantly making it impossible to maneuver around. Kind of hard to fly when every touch of a button makes you shoot out like a rocket

**Comment by tsunamayo** [comment]
Yes it seems more like a problem of thrust / weight calibration for now and acceleration value. I will definitely address this for sure as we balance the ship. How big is your ship? That could be cool if you could send me the blueprint. Thanks!

**Comment by ZachZent** [comment]
https://cdn.discordapp.com/attachments/406664114249793536/666008124913745970/Janoi_Carrier_Fighter.mps

This is a problem with all ships really, but especially small ones. This ones flies at around 1k km/hr which compared to others is on the slower side. Yes the problem with thrust/weight calibration defiantly needs to be addressed sooner then later, but a speed control would be a welcome addition as well. For example landing on a carrier. Would be difficult to maneuver in 6dof when you leap up to max speed regardless of the ship.

As for my fighter, I think you will find it interesting. Has a fully auto gun at the press of a button.

**Comment by Dwarf-LordPangolin** [comment]
I concur with Zach. While the thrust/weight ratios do indeed need to be calibrated (ships are very zippy right now! :D ), these are two separate issues. Even with the limited amount of testing I've done so far vs NPC ships, I can tell that one of the most-needed features is throttle control. Combat is frustrating without it, particularly if there's a significant difference between the speeds of two craft. Without it, it's impossible to stay behind an opponent if you manage to get there in a dogfight; you either keep running into their aft, or having to break off and lose your good position.


A couple suggestions, if you're worried about the control scheme becoming too complicated:
**Boost:** the W and S keys aren't being used for anything else during Boost mode, so holding them down to adjust the throttle up and down would be very intuitive for players.

**6dof:** as Zach notes, it's impossible to maneuver near other ships with the high acceleration we have right now. Adjusting the thrust/weight ratios won't solve this, as we are going to want the same ship to have high 6dof acceleration while in combat, and low 6dof acceleration outside of combat. Although a pair of dedicated throttle keys could do this, that would make the control scheme more complicated, and would also require the player to move their hand more than is convenient.

A simpler and more ergonomic solution would be to bind one of the modifier keys (maybe Alt?) for use in adjusting the throttle when combined with W and S. Alt+S would decrease the 6dof throttle, Alt+W would increase it. When not pressing Alt, W and S would accelerate forward and back, just like they currently do.

So, someone getting ready for a fight in 6dof would hold Alt+W until their throttle said 100%, but if they were getting ready to find a good landing spot on a planet, they would hold Alt+S until their throttle was something a bit slower and less likely to send them rocketing into a lake. ;)

**Comment by tsunamayo** [comment]
Okay as a quick solution I will implement the throttle on the boost mode as this one makes sense.
For the other one, first I want to work on calibration, I will lower the acceleration which should help tremendously. I will also change my current formula (LN(1 + thrustToWeight), to something capped, so I can define a maximum speed / acceleration. 
I keep you guy posted, this is obviously very important and sensitive, but thats why we are testing!
Cheers

**Comment by tgiddings** [comment]
It seems like this is about tuning how much of a ship's acceleration is actually used. If that is the case, I think it would be desirable to tie it into the logic system. This would open the door to things like switching between a precision mode and a get-out-fast mode.

**Comment by Kaiser-Indrasil** [comment]
> It seems like this is about tuning how much of a ship's acceleration is actually used. If that is the case, I think it would be desirable to tie it into the logic system. This would open the door to things like switching between a precision mode and a get-out-fast mode.

No, this is actually about controlling the ship's speed rather than controlling child entity's acceleration.

**Comment by tgiddings** [comment]
> No, this is actually about controlling the ship's speed rather than controlling child entity's acceleration.

I never mentioned child entities.

I guess I misremembered this games physics -- I thought it was Newtonian, meaning top speed is only a matter of gameplay limitations.

For Tuning top speed, I think tying into the logic system is still desirable -- for the same reasons.

**Comment by Dwarf-LordPangolin** [comment]
>No, this is actually about controlling the ship's speed rather than controlling child entity's acceleration.

I think when tgiddings said "open the door to" he was using it in the figurative sense, not talking about an actual door.

To recap, the main reason control over the speed is required is simply to prevent the player from ramming into things because they're going too fast -- in particular, to keep them from ramming into other ships that they're trying to fight. If the ship I'm fighting has a top speed of 200 m/sec, but my ship has a top speed of 300 m/sec, **the game won't be fun** -- because the entire fight will consist of me briefly managing to point my guns at the target to get a few shots off before either whizzing past it because I can't slow down, or ramming into its backside.

Currently, trying to fly without throttle control makes the game feel like the film _Speed_. 😉 

I would go beyond tying it into the logic system and make it as fundamental to the game as the movement keys are. Precise control over movement is something players reasonably expect from games like this.

**Comment by tgiddings** [comment]
> I would go beyond tying it into the logic system and make it as fundamental to the game as the movement keys are.

Please elaborate on what you mean here. I have seen speed done a lot of ways in a lot of games, and while I've seen ways that are very good or very bad for any given game, I've never seen something that felt fundamental. Even the throttle-based maneuverability in Elite: Dangerous doesn't feel fundamental to me.
